frontend/src/test/scala/bloop/bsp/BspProtocolSpec.scala

```diff
@@ -507,6 +571,11 @@ class BspProtocolSpec {
     }
 
     val startedTask = scala.collection.mutable.HashSet[bsp.TaskId]()
+    type DiagnosticKey = (bsp.BuildTargetIdentifier, Int) // Int = Cycle starts at 0
+    val stringifiedDiagnostics = new ConcurrentHashMap[DiagnosticKey, StringBuilder]()
```

Comment:
I recommend giving `TrieMap`  a try since it has non-`AnyRef` parameters types for key lookups.

Reply:
I use `ConcurrentHashMap` because `compute` allows me to easily aggregate strings in the case where there already exists a string builder and the case it doesn't. I tried to play with the `TrieMap` interface but didn't find anything equivalent to `compute`